### ISSUE TITLE
ulid_date() should accept timezones

### ISSUE THREAD

**Issue by worthmine** [bk/Data-ULID#3]
Hi, I've been interested in this module today.
And [tried to fix](https://github.com/worthmine/Data-ULID/tree/dev) the inconvenience when I use it. 

If you like this I'll try to make a PR with [the least changes](https://github.com/bk/Data-ULID/compare/master...worthmine:master).

the diff is here. 
```diff
sub ulid_date {
-    my $ulid = shift;
+    my $ulid     = shift;
+    my %args     = @_ if @_ >= 2;
+    my $timezone = shift || 'UTC';
+    $timezone = $args{'time_zone'} if exists $args{'time_zone'};
    die "ulid_date() needs a normal or binary ULID as parameter" unless $ulid;
-    my ($ts, $rand) = _ulid($ulid);
+    my ( $ts, $rand ) = _ulid($ulid);
    $ts = _bint($ts);
-    return DateTime->from_epoch(epoch=>$ts / 1000);
+    return DateTime->from_epoch( epoch => $ts / 1000, time_zone => $timezone );
}
```
I've just added to accept timezones(and some tidied)

best regard.

**Comment by bk** [comment]
Thanks for your contribution, @worthmine. This is certainly an interesting idea, but I will need some time to consider whether I think it is a good one. You can always call `set_time_zone()` on the `DateTime` object which is returned by `ulid_date()`, so the convenience of setting it inside the function is minimal. And it might actually be confusing to add this functionality, given that the time stamps encoded by the ULIDs generated by this module are always UTC.

**Comment by worthmine** [comment]
Thank you for reviewing!

Ok, I agree with your stance.

but EVEN IF you think you don’t have to change the spec that `ulid_date()` always returns UTC DateTime, you have to change something strange.

Because the timestamp strings returned by this module should include 'Z' at the end of strings for declaring "the offset is UTC", but not. This is required on ISO8601, RFC is [3339](https://tools.ietf.org/html/rfc3339#section-4.2)
So normally, we,(not only human, but including the machines) will consider that they are local timestamps.

That’s the point of I wanna fix.

However, after I wrote the above, I thought it may be a mis-spec of [DateTime](https://metacpan.org/pod/DateTime#$dt-%3Eset_time_zone(-$tz-)) that returns the timestamp without ending 'Z' with setting 'UTC’.

This module just depends on that so if that is fixed,  you may not have to fix.

**Comment by bk** [comment]
What is returned by `ulid_date()` is a `DateTime` object, not a string, so it does not have a fixed representation as a string. 

The default stringification of the `DateTime` object consists in calling its `datetime()` method, which does not include the time zone – something that would not be changed by setting the time zone explicitly. This omission is arguably a mistake by the module authors. To get an ISO8601 representation you have to do something similar to `$dt->strftime('%FT%T%z')`. This yields a string like `2020-10-17T21:05:10+0000`, which is formally speaking according to spec although you almost always would see a colon between the hours and minutes parts of the zone indicator: `2020-10-17T21:05:10+00:00`.

So basically my answer is that yes, there is a bit of a problem here – but it consists in an inconvenience in formatting `DateTime` objects, which is something that lies outside the scope of my module. Therefore I think that I will close this issue. But thanks for bringing this up – it's been very interesting to look into this matter.
